yt_dlp/networking/_urllib.py

```diff
@@ -341,7 +341,7 @@ def redirect_request(self, req, fp, code, msg, headers, newurl):
         new_method = get_redirect_method(m, code)
 
         # only remove payload if method changed (e.g. POST to GET)
-        if new_method != m and new_data is not None:
+        if new_method != m and new_method == 'GET':
```

Comment:
HEAD too, right?

Reply:
nothing with a payload changes to HEAD, only GET

but thinking now this can probably be simplified to just `new_method != m`